palette/src/rgb/rgb.rs

```diff
@@ -738,6 +742,36 @@ where
     }
 }
 
+impl<S, T> FromColorUnclamped<Oklab<T>> for Rgb<S, T>
+where
+    T: Real + Powi + Arithmetics + FromScalar + Copy,
+    T::Scalar: Recip
+        + IsValidDivisor<Mask = bool>
+        + Arithmetics
+        + FromScalar<Scalar = T::Scalar>
+        + Real
+        + Zero
+        + One
+        + Clone,
+    S: RgbStandard + 'static,
+    S::TransferFn: FromLinear<T, T>,
+    S::Space: RgbSpace<WhitePoint = D65>,
+    <S::Space as RgbSpace>::Primaries: Primaries<T::Scalar>,
+    Yxy<Any, T::Scalar>: IntoColorUnclamped<Xyz<Any, T::Scalar>>,
+{
+    fn from_color_unclamped(oklab: Oklab<T>) -> Self {
+        if TypeId::of::<<S as RgbStandard>::Space>() == TypeId::of::<Srgb>() {
+            // Use direct sRGB to Oklab conversion
+            // fixme: why does the direct conversion produce relevant differences
+            // to the conversion via XYZ?
```

Comment:
Its entirely possible, that it is just due to rounding errors, but I don't have more than a guess here, either.

Reply:
That's likely a contributing factor. The conversion via XYZ doesn't use pre-defined matrices (yet), so that's a difference I know of.

Well, I'm happy that it's better, regardless. I think this bit of the comment can be removed. I also prefer to have remaining todos/fixmes in the issue tracker, rather than in code, so I would like to make sure there are none left.

Reply:
>  I also prefer to have remaining todos/fixmes in the issue tracker, rather than in code, so I would like to make sure there are none left.

There are two issues, that I can see in the TODOs/FIXMEs
1. The range of colors created for distributions of Oklab and Oklch. We should resolve this before we publish the PR. And the FIXMEs/TODOs won't be necessary then.
2. The erroneous conversion using f32 on some CPUs (or maybe just on my machine, maybe all AMD CPUs) , that I need @bottosson 's guidance for. This has one FIXME/TODO in the code and two in the tests. I think they need to stay in the tests to avoid a misleading appearance of the tests and to be able to find the relevant tests, if and when we find a way to correct the error. As long as there is no fix, maybe we should elevate the FIXME to a warning in the documentation or even a trait bound, that allows only `f64`, as the error might be catastrophic for some library users.

Reply:
Yes we definitely need to do something about point 1. In the (likely, it seems) case that we don't get a response for point 2, we should add a note in the documentation about it being sensitive to precision. It could be that it's not important for everyone, but it's a good idea to be a bit proactive with a warning that explains the phenomenon a bit.